linkerd/app/core/src/classify.rs

```diff
@@ -127,19 +123,36 @@ impl classify::ClassifyResponse for Response {
     type ClassifyEos = Eos;
 
     fn start<B>(self, rsp: &http::Response<B>) -> Eos {
+        let status = rsp.status();
         match self {
-            Response::Default => grpc_class(rsp.headers())
-                .map(|c| Eos::Grpc(GrpcEos::NoBody(c)))
-                .unwrap_or_else(|| Eos::Default(rsp.status())),
-            Response::Grpc => grpc_class(rsp.headers())
-                .map(|c| Eos::Grpc(GrpcEos::NoBody(c)))
-                .unwrap_or(Eos::Grpc(GrpcEos::Open)),
-            Response::Profile(ref classes) => Self::match_class(rsp, classes.as_ref())
-                .map(Eos::Profile)
+            Self::Http(statuses) => Eos::Class(Class::Http(if statuses.contains(status) {
+                Err(status)
+            } else {
+                Ok(status)
+            })),
```

Comment:
nit, take it or leave it: would it make sense for `http::StatusRanges` and `grpc::Codes` to have a function that takes a status/code and returns a `Result<Status>`/`Result<Code>`, rather than just a `bool`? we repeat the if statement here in a few places...